domain/application/state/application.go

```diff
@@ -806,6 +807,63 @@ func (st *State) GetApplicationLifeByName(ctx context.Context, appName string) (
 	return app.UUID, app.LifeID, errors.Capture(err)
 }
 
+// CheckAllApplicationsAndUnitsAreAlive checks that all applications and units
+// in the model are alive, returning an error if any are not.
+// The following errors may be returned:
+// - [applicationerrors.ApplicationNotAlive] if any applications are not alive.
+// - [applicationerrors.UnitNotAlive] if any units are not alive.
+func (st *State) CheckAllApplicationsAndUnitsAreAlive(ctx context.Context) error {
+	db, err := st.DB()
+	if err != nil {
+		return errors.Capture(err)
+	}
+
+	checkApplicationsStmt, err := st.Prepare(`
+SELECT &applicationName.*
+FROM application
+WHERE life_id != 0
+ORDER BY name
```

Comment:
Why are these `ORDER BY` necessary? It looks they are only used for crafting the error, where the order won't matter.

Reply:
Yes that is true